Source/Core/Core/IOS/WFS/WFSSRV.cpp

```diff
@@ -118,23 +118,77 @@ std::optional<IPCReply> WFSSRVDevice::IOCtl(const IOCtlRequest& request)
     break;
   }
 
-  // TODO(wfs): Globbing is not really implemented, we just fake the one case
-  // (listing /vol/*) which is required to get the installer to work.
+  // TODO(wfs): Handle more patterns (e.g. with '?')
   case IOCTL_WFS_GLOB_START:
-    INFO_LOG_FMT(IOS_WFS, "IOCTL_WFS_GLOB_START({})", request.request);
-    Memory::Memset(request.buffer_out, 0, request.buffer_out_size);
-    Memory::CopyToEmu(request.buffer_out + 0x14, m_device_name.data(), m_device_name.size());
+  {
+    const std::string pattern = NormalizePath(
+        Memory::GetString(request.buffer_in + 0x2, Memory::Read_U16(request.buffer_in)));
+    const size_t size = pattern.size();
+    if (size < 2 || pattern[size - 1] != '*' || pattern[size - 2] != '/')
```

Comment:
Is this part based on reverse engineering of DQX's IOS?

Reply:
Yes, all the code is based on static analysis of the IOS59 module. Ideally the root of the WFS folder would correspond to `/vol/msc01` but we can't do that anymore because that would break backwards compatibility. That means some paths need special handling: `/vol` but also `/dev` based on the error above and maybe `/wfs` too.